docs/interpreter/scio.md

```diff
@@ -0,0 +1,150 @@
+limitations under the License.
+-->
+{% include JB/setup %}
+
+# Scio Interpreter for Apache Zeppelin
+
+<div id="toc"></div>
+
+## Overview
+Scio is a Scala DSL for [Google Cloud Dataflow](https://github.com/GoogleCloudPlatform/DataflowJavaSDK) and [Apache Beam](http://beam.incubator.apache.org/) inspired by [Spark](http://spark.apache.org/) and [Scalding](https://github.com/twitter/scalding). See the current [wiki](https://github.com/spotify/scio/wiki) and [API documentation](http://spotify.github.io/scio/) for more information.
+
+## Configuration
+<table class="table-configuration">
+  <tr>
+    <th>Name</th>
+    <th>Default Value</th>
+    <th>Description</th>
+  </tr>
+  <tr>
+    <td>zeppelin.scio.argz</td>
+    <td>--runner=InProcessPipelineRunner</td>
+    <td>Scio interpreter wide arguments</td>
+  </tr>
+  <tr>
+    <td>zeppelin.scio.maxResult</td>
+    <td>1000</td>
+    <td>Max number of SCollection results to display</td>
+  </tr>
+
+</table>
+
+## Enabling the Scio Interpreter
+
+In a notebook, to enable the **Scio** interpreter, click the **Gear** icon and select **scio**.
+
+## Using the Scio Interpreter
+
+In a paragraph, use `%scio` to select the **Scio** interpreter. You can use it much the same way as vanilla Scala REPL and [Scio REPL](https://github.com/spotify/scio/wiki/Scio-REPL). Context is shared among all *Scio* paragraphs. There is a special variable **argz** which holds arguments from Scio interpreter settings. The easiest way to proceed is to create a context via standard `ContextAndArgs`.
+
+```scala
+%scio
+val (sc, args) = ContextAndArgs(argz)
+```
+
+Use `sc` context the way you would in regular pipeline/REPL.
+
+Example:
+
+```scala
+%scio
+val (sc, args) = ContextAndArgs(argz)
+sc.parallelize(Seq("foo", "foo", "bar")).countByValue.closeAndDisplay()
+```
+
+Please refer to [Scio wiki](https://github.com/spotify/scio/wiki) for more complex examples.
+
+### Progress
+
+There can be only one paragraph running at a time. There is no notion of overall progress, thus progress bar will be `0`.
+
+### SCollection display helpers
+
+Scio interpreter comes with display helpers to ease working with Zeppelin notebooks. Simply use `closeAndDisplay()` on `SCollection` to close context and display the results. The number of results is limited by `zeppelin.scio.maxResult` (by default 1000).
+
+Supported `SCollection` types:
+
+ * Scio's typed BigQuery
+ * Scala's Products (case classes, tuples)
+ * Google BigQuery's TableRow
+ * Apache Avro
+ * All Scala's `AnyVal`
+
+#### BigQuery example:
+
+```scala
+%scio
+@BigQueryType.fromQuery("""|SELECT departure_airport,count(case when departure_delay>0 then 1 else 0 end) as no_of_delays
+                           |FROM [bigquery-samples:airline_ontime_data.flights]
+                           |group by departure_airport
+                           |order by 2 desc
+                           |limit 10""".stripMargin) class Flights
+
+val (sc, args) = ContextAndArgs(argz)
```

Comment:
If sc is created every time in each paragraph ? How is it shared between paragraphs. ?